opencog/query/Implicator.h

```diff
@@ -47,20 +47,31 @@ namespace opencog {
  *
  * The 'var_soln' argument in the callback contains the map from variables
  * to ground terms. 'class Instantiator' is used to perform the actual
- * grounding.  A list of grounded expressions is created in 'result_list'.
+ * grounding.  A set of grounded expressions is created in 'result_set'.
+ * Note that the callback may be called many times reporting the same
+ * results. In that case the 'result_set' will contain unique solutions.
  */
 class Implicator :
 	public virtual PatternMatchCallback
 {
+	protected:
+		UnorderedHandleSet _result_set;
+		HandleSeq _result_list;
```

Comment:
I wanted to avoid copying from unordered set each time, but did it overcomplicated here:
https://github.com/opencog/atomspace/blob/master/opencog/query/Implicator.cc#L77-L78